etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java

```diff
@@ -131,4 +138,16 @@ DomainResponse handleOrders(DomainRequest request) {
         OrderResponse orderResponse = new OrderResponse(orders);
         return domainResponseHelper.constructOkResponse(orderResponse);
     }
+
+    DomainResponse handleMetadata(DomainRequest request) {
+        try {
+            String uniqueId = request.getBody();
+            PartnerMetadata metadata = partnerMetadataStorage.readMetadata(uniqueId);
+            return domainResponseHelper.constructOkResponse(
+                    formatter.convertToJsonString(metadata));
+        } catch (PartnerMetadataException | FormatterProcessingException e) {
+            logger.logError("Unable to read metadata", e);
+            return domainResponseHelper.constructErrorResponse(400, e);
```

Comment:
This should be a `500`, you think?  Both those exceptions are generated when bad stuff happens and there's nothing a caller can do about it, right?

Reply:
Right! Good call

Reply:
Fixed